### PR TITLE
[Rest Api Compatibility] Enable tests that are already fixed

### PR DESCRIPTION
**Pull request by pgomulka**
With types removal changes being available under rest api compatibility I have removed the block entries for tests which are already fixed

### CODE REVIEW

File: modules/ingest-user-agent/build.gradle [hunk 1]
```diff
@@ -26,7 +26,6 @@ testClusters.all {
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
     'ingest-useragent/30_custom_regex/Test user agent processor with custom regex file',
-    'ingest-useragent/20_useragent_processor/Test user agent processor with defaults',
```
Comment (pgomulka) [inline modules/ingest-user-agent/build.gradle:29]: indexing with type was used which is fixed now

File: modules/parent-join/build.gradle [hunk 2]
```diff
@@ -22,8 +22,6 @@ restResources {
 
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
-    '/20_parent_join/Test parent_id query',
```
Comment (pgomulka) [inline modules/parent-join/build.gradle:25]: search was returing `_type` and it is fixed now

File: modules/lang-painless/build.gradle [hunk 3]
```diff
@@ -51,18 +51,6 @@ tasks.named("test").configure {
 
 tasks.named("yamlRestCompatTest").configure {
     systemProperty 'tests.rest.blacklist', [
-            'painless/70_execute_painless_scripts/Execute with date field context (single-value)',
```
Comment (pgomulka) [inline modules/lang-painless/build.gradle:54]: these were not type related. Those were added temporarily before backporting the changes to 7.x
https://github.com/elastic/elasticsearch/pull/72135/files#r619034432

**Review by joegallo — APPROVED** [review]
Easy +1 -- ship it. 😄

**Pull request merged**